drivers/pwm/pwm_max31790.c

```diff
@@ -0,0 +1,690 @@
+				   MAX37190REGISTER_FANXCONFIGURATION_LOCKEDROTORPOLARITY_POS,     \
+				   MAX37190REGISTER_FANXCONFIGURATION_LOCKEDROTORPOLARITY_LENGTH)
+#define MAX37190REGISTER_FANXCONFIGURATION_TACH_LENGTH 1
+#define MAX37190REGISTER_FANXCONFIGURATION_TACH_POS    0
+#define MAX37190REGISTER_FANXCONFIGURATION_TACH_GET(value)                                         \
+	MAX37190REGISTER_GET_VALUE(value, MAX37190REGISTER_FANXCONFIGURATION_TACH_POS,             \
+				   MAX37190REGISTER_FANXCONFIGURATION_TACH_LENGTH)
+#define MAX37190REGISTER_FANXCONFIGURATION_TACH_SET(target, value)                                 \
+	MAX37190REGISTER_SET_VALUE(target, value, MAX37190REGISTER_FANXCONFIGURATION_TACH_POS,     \
+				   MAX37190REGISTER_FANXCONFIGURATION_TACH_LENGTH)
+#define MAX37190REGISTER_FANXDYNAMICS_SPEEDRANGE_LENGTH 3
+#define MAX37190REGISTER_FANXDYNAMICS_SPEEDRANGE_POS	5
+#define MAX37190REGISTER_FANXDYNAMICS_SPEEDRANGE_GET(value)                                        \
+	MAX37190REGISTER_GET_VALUE(value, MAX37190REGISTER_FANXDYNAMICS_SPEEDRANGE_POS,            \
+				   MAX37190REGISTER_FANXDYNAMICS_SPEEDRANGE_LENGTH)
+#define MAX37190REGISTER_FANXDYNAMICS_SPEEDRANGE_SET(target, value)                                \
+	MAX37190REGISTER_SET_VALUE(target, value, MAX37190REGISTER_FANXDYNAMICS_SPEEDRANGE_POS,    \
+				   MAX37190REGISTER_FANXDYNAMICS_SPEEDRANGE_LENGTH)
+#define MAX37190REGISTER_FANXDYNAMICS_PWMRATEOFCHANGE_LENGTH 3
+#define MAX37190REGISTER_FANXDYNAMICS_PWMRATEOFCHANGE_POS    2
+#define MAX37190REGISTER_FANXDYNAMICS_PWMRATEOFCHANGE_GET(value)                                   \
+	MAX37190REGISTER_GET_VALUE(value, MAX37190REGISTER_FANXDYNAMICS_PWMRATEOFCHANGE_POS,       \
+				   MAX37190REGISTER_FANXDYNAMICS_PWMRATEOFCHANGE_LENGTH)
+#define MAX37190REGISTER_FANXDYNAMICS_PWMRATEOFCHANGE_SET(target, value)                           \
+	MAX37190REGISTER_SET_VALUE(target, value,                                                  \
+				   MAX37190REGISTER_FANXDYNAMICS_PWMRATEOFCHANGE_POS,              \
+				   MAX37190REGISTER_FANXDYNAMICS_PWMRATEOFCHANGE_LENGTH)
+#define MAX37190REGISTER_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_LENGTH 1
+#define MAX37190REGISTER_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_POS    1
+#define MAX37190REGISTER_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_GET(value)                            \
+	MAX37190REGISTER_GET_VALUE(value,                                                          \
+				   MAX37190REGISTER_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_POS,       \
+				   MAX37190REGISTER_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_LENGTH)
+#define MAX37190REGISTER_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_SET(target, value)                    \
+	MAX37190REGISTER_SET_VALUE(target, value,                                                  \
+				   MAX37190REGISTER_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_POS,       \
+				   MAX37190REGISTER_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_LENGTH)
```

Comment:
Please find/use  shorter names, at least REGISTER can be dropped.

Reply:
I would like to keep the controller name in there, to avoid name conflicts. On top of that I would also like to keep the register name and the bit(s)-name, hence I cannot really remove more than register. But I dropped the register part, makes it at least a little bit shorter.